src/pages/PullRequestPage/PullCoverage/routes/IndirectChangesTab/IndirectChangedFiles/hooks/useIndirectChangedFilesTable.ts

```diff
@@ -0,0 +1,165 @@
+import { SortingState } from '@tanstack/react-table'
+import { isArray } from 'lodash'
+import isNumber from 'lodash/isNumber'
+import qs, { ParsedQs } from 'qs'
+import { useMemo, useState } from 'react'
+import { useLocation, useParams } from 'react-router-dom'
+
+import { PullSchemaType, usePull } from 'services/pull'
+
+const orderingDirection = Object.freeze({
+  desc: 'DESC',
+  asc: 'ASC',
+})
+
+const orderingParameter = Object.freeze({
+  name: 'FILE_NAME',
+  change: 'CHANGE_COVERAGE',
+  patch: 'PATCH_COVERAGE',
+  head: 'HEAD_COVERAGE',
+  missesCount: 'MISSES_COUNT',
+})
+
+function getFilters({
+  sorting,
+  flags,
+  components,
+}: {
+  sorting: Array<{ id: string; desc: boolean }>
+  flags?: ParsedQs[] | string[]
+  components?: ParsedQs[] | string[]
+}) {
+  const state = sorting.at(0)
+  const direction = state?.desc ? orderingDirection.desc : orderingDirection.asc
+
+  let parameter = undefined
+  if (state && state.id === 'name') {
+    parameter = orderingParameter.name
+  }
+
+  if (state && state.id === 'missedLines') {
+    parameter = orderingParameter.missesCount
+  }
+
+  if (state && state.id === 'head') {
+    parameter = orderingParameter.head
+  }
+
+  if (state && state.id === 'change') {
+    parameter = orderingParameter.change
+  }
+
+  return {
+    ordering: {
+      direction,
+      parameter,
+    },
+    hasUnintendedChanges: true,
+    ...(flags ? { flags } : {}),
+    ...(components ? { components } : {}),
+  }
+}
+
+function transformIndirectChangesData(pull: PullSchemaType | null | undefined) {
```

Comment:
Thoughts on typing out the return objects for the functions here?

Reply:
I feel it makes the code more verbose and creates the impression we want to be strict about the return value of this function. We might need to send more info with future changes so I prefer to keep this flexible for now. 

The type is inferred so most IDEs should correct any misuse here.

Reply:
I don't think strict returns now and flexible later is necessarily an issue nor mutually exclusive. But I get your point here, it's more of a stylistic thing